### PR TITLE
Add support for Azure availability sets (refactored)

### PR DESCRIPTION
**Pull request by sk4zuzu**
This is an improved version of #1411. All initial commits are included. Comments from code reviews have been addressed. Simple schema validation and changelog entry have been added.

### CODE REVIEW

**Comment by sk4zuzu** [comment]
@sunshine69 please take a look at this one. :+1:

File: docs/home/howto/CLUSTER.md [hunk 1]
```diff
@@ -649,3 +649,89 @@ You can read more [here](https://www.confluent.io/blog/how-choose-number-topics-
...
+specification:
+# The "name" attribute is generated automatically according to Epiphany's naming conventions
+  availability_set: kube-node  # Short and simple name is preferred
+  platform_fault_domain_count: 2
+  platform_update_domain_count: 5
+  managed: 'true'
+provider: azure
+```
+
+Then in the corresponding `components` section of the kind `kind: epiphany-cluster` set it as well.
```
Comment (to-bar) [inline docs/home/howto/CLUSTER.md:671]:
```suggestion
Then set it also in the corresponding `components` section of the `kind: epiphany-cluster` doc.
```

File: docs/home/howto/CLUSTER.md [hunk 2]
```diff
@@ -649,3 +649,89 @@ You can read more [here](https://www.confluent.io/blog/how-choose-number-topics-
...
+      count: 1
+    kubernetes_node:
+# This line tells we generate the availability-set terraform template
+      availability_set: kube-node  # Short and simple name is preferred
+      count: 2
+```
+
+The full minimum cluster is listed below. Note that you can set different type of machine within different
+availability set, eg. all kubenetes master in one availability set, all kubernetes nodes in another one, also
+kafka etc.. .
```
Comment (to-bar) [inline docs/home/howto/CLUSTER.md:687]:
```suggestion
The full minimum cluster is listed below. Note that you can set different type of machine within different
availability set, e.g. all `kubernetes_master` in one availability set, all `kubernetes_node` in another one, also
`kafka` etc.
```

**Review by to-bar — CHANGES_REQUESTED** [review]

**Review by to-bar — APPROVED** [review]

File: core/src/epicli/data/azure/validation/infrastructure/availability-set.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  - managed
+properties:
+  availability_set:
+    "$id": "#/properties/specification/properties/availability_set"
+    type: string
+    pattern: ^[a-z0-9\-_]+$
+  platform_fault_domain_count:
+    "$id": "#/properties/specification/properties/platform_fault_domain_count"
+    type: integer
+    minimum: 1
```
Comment (to-bar) [inline core/src/epicli/data/azure/validation/infrastructure/availability-set.yml:14]:
```suggestion
    minimum: 2
```

File: core/src/epicli/data/azure/validation/infrastructure/availability-set.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  platform_update_domain_count:
+    "$id": "#/properties/specification/properties/platform_update_domain_count"
+    type: integer
+    minimum: 1
```
Comment (to-bar) [inline core/src/epicli/data/azure/validation/infrastructure/availability-set.yml:18]:
```suggestion
    minimum: 2
```

File: core/src/epicli/data/azure/validation/infrastructure/availability-set.yml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  managed:
+    "$id": "#/properties/specification/properties/managed"
+    type: string
```
Comment (to-bar) [inline core/src/epicli/data/azure/validation/infrastructure/availability-set.yml:21]: Can we accept (switch to) YAML bool type for consistency with other config docs? https://yaml.org/type/bool.html

**Review by to-bar — CHANGES_REQUESTED** [review]

File: core/src/epicli/data/azure/validation/infrastructure/availability-set.yml (hunk 5 above)
Comment (sk4zuzu) [reply core/src/epicli/data/azure/validation/infrastructure/availability-set.yml:21]: Yes. I'll work on that!

File: core/src/epicli/data/azure/defaults/infrastructure/availability-set.yml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+kind: infrastructure/availability-set
+title: "Availability Set Config"
+provider: azure
+name: default
+specification:
+    name: SET_BY_AUTOMATION
+    platform_fault_domain_count: SET_BY_AUTOMATION
+    platform_update_domain_count: SET_BY_AUTOMATION
+    managed: SET_BY_AUTOMATION
```
Comment (to-bar) [inline core/src/epicli/data/azure/defaults/infrastructure/availability-set.yml:9]:
```suggestion
    platform_fault_domain_count: 2
    platform_update_domain_count: 5
    managed: true
```

**Review by to-bar — CHANGES_REQUESTED** [review]

File: docs/home/howto/CLUSTER.md [hunk 7]
```diff
@@ -669,3 +669,97 @@ You can read more [here](https://www.confluent.io/blog/how-choose-number-topics-
...
+      count: 1
+    kubernetes_node:
+# This line tells we generate the availability-set terraform template
+      availability_set: kube-node  # Short and simple name is preferred
+      count: 2
+```
+
+The full minimum cluster is listed below. Note that you can set different type of machine within different
+availability set, e.g. all `kubernetes_master` in one availability set, all `kubernetes_node` in another one, also
+`kafka` etc.
```
Comment (to-bar) [inline docs/home/howto/CLUSTER.md:706]:
```suggestion
The example below shows a complete configuration. Note that it's recommended to have a dedicated availability set for each clustered component.
```

**Review by to-bar — CHANGES_REQUESTED** [review]

**Review by to-bar — APPROVED** [review]

**Comment by mkyc** [comment]
@sunshine69 it would be good if you could have a look at this PR.

**Review by atsikham — APPROVED** [review]
LGTM
Good to know from this PR that there is a parameters validation part. @sk4zuzu does it documented somewhere or just exist in the code?

**Comment by sk4zuzu** [comment]
> LGTM
> Good to know from this PR that there is a parameters validation part. @sk4zuzu does it documented somewhere or just exist in the code?

I don't know such place. I just kinda reverse-engineered for my own PRs and started using it. 🤭 I think the general feel is that this particular solution is not ideal and probably needs to be improved, but for the time being we can use it, it's better than nothing IMHO :+1:

**Pull request merged**